Kernel/Arch/aarch64/RPi/MMIO.cpp

```diff
@@ -4,16 +4,16 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
-#include <Kernel/Arch/aarch64/MainIdRegister.h>
+#include <Kernel/Arch/aarch64/CPUID.h>
 #include <Kernel/Arch/aarch64/RPi/MMIO.h>
 
 namespace Kernel::RPi {
 
 MMIO::MMIO()
     : m_base_address(0xFE00'0000)
 {
-    MainIdRegister id;
-    if (id.part_num() <= MainIdRegister::RaspberryPi3)
+    auto main_id_register = Aarch64::MIDR_EL1::read();
```

Comment:
Why is this better than what was here before? The code's still device-dependent (cf 0x3f00'0000), and on the LHS the checking code is device-dependent too. That's imho clearer (?)

Reply:
This is yet another instance of me messing up the pull requests, this belongs to #17201; Iʼm pretty convinced itʼs a decent change and Iʼll gladly explain the my reasoning there! 🙃